packages/dev/core/src/Cameras/Inputs/arcRotateCameraMouseWheelInput.ts

```diff
@@ -210,8 +210,12 @@ export class ArcRotateCameraMouseWheelInput implements ICameraInput<ArcRotateCam
         // a vector defining where we want to zoom to.
         const ray = scene.createPickingRay(scene.pointerX, scene.pointerY, Matrix.Identity(), camera, false);
         // Since the camera is the origin of the picking ray, we need to offset it by the camera's offset manually
-        ray.origin.x -= camera.targetScreenOffset.x;
-        ray.origin.y -= camera.targetScreenOffset.y;
+        // Because the offset is in view space, we need to convert it to world space first
+        const viewOffset = new Vector3(camera.targetScreenOffset.x, camera.targetScreenOffset.y, 0);
+        camera.getViewMatrix().invertToRef(camera._cameraTransformMatrix);
+        const globalOffset = Vector3.TransformNormal(viewOffset, camera._cameraTransformMatrix);

```

Comment:
TransformNormal requires a normalized vector

Reply:
The documentation and name of `TransformNormal` is quite misleading, actually. It transforms vectors, not normals, so the parameter does not have to be normalized.

Here, we do need the parameter not to be normalized, because we want to transform the (x,y) offset from view space to world space.

Reply:
I guess the doc is misleading as well then:

`This methods computes transformed normalized direction vectors only (...)`

Reply:
Resolving this.

Reply:
I was going to, but I'll leave you to it then ;)